### PR TITLE
fix(deisctl): fixes for graceful upgrades.

### PR DESCRIPTION
**Pull request by iancoffey**
- fix stateless upgrade handling
- stop extra router cycling

### CODE REVIEW

**Comment by iancoffey** [comment]
fixes #4419
fixes #4503

File: deisctl/client/client.go [hunk 1]
```diff
@@ -74,13 +74,19 @@ func (c *Client) UpgradePrep(argv []string) error {
 	usage := `Prepare platform for graceful upgrade.
 
 Usage:
-  deisctl upgrade-prep [options]
+  deisctl upgrade-prep [--stateless]
```
Comment (adamreese) [inline deisctl/client/client.go]: Could we check to see if store services are running to know if they need to be upgraded?

File: deisctl/cmd/upgrade.go [hunk 2]
```diff
@@ -65,16 +69,15 @@ func republishServices(ttl uint64, nodes []*model.ConfigNode, cb config.Backend)
 }
 
 // UpgradeTakeover gracefully starts a platform stopped with UpgradePrep
-func UpgradeTakeover(b backend.Backend, cb config.Backend) error {
-
-	if err := doUpgradeTakeOver(b, cb); err != nil {
+func UpgradeTakeover(stateless bool, b backend.Backend, cb config.Backend) error {
```
Comment (adamreese) [inline deisctl/cmd/upgrade.go]: It would be nice to have the ability to specify version upgrading to and validate unit files are desired version.

**Comment by michelleN** [comment]
:+1:

File: deisctl/client/client.go (hunk 1 above)
Comment (iancoffey) [inline deisctl/client/client.go]: Yeah, auto-detect could be possible down the road. I personally like the explicit flag for now.

File: deisctl/cmd/upgrade.go (hunk 2 above)
Comment (iancoffey) [inline deisctl/cmd/upgrade.go]: Rolling in a unit refresh has been brought up a few times, it does make sense. For now, I just like allowing the user to use whatever units they want along with documenation, and having a little less machinery in deisctl.

**Pull request merged**